src/imap/scan_folders.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Instant;
+
+use crate::{config::Config, context::Context, dc_tools::time};
+use anyhow::Context as _;
+
+use crate::error::Result;
+use crate::imap::Imap;
+use async_std::prelude::*;
+
+use super::{get_folder_meaning, get_folder_meaning_by_name, FolderMeaning};
+
+impl Imap {
+    pub async fn scan_folders(&mut self, context: &Context) -> Result<()> {
+        use crate::config::Config::*;
+
+        // First of all, debounce to once per minute:
+        let mut last_scan = context.last_full_folder_scan.lock().await;
+        if let Some(last_scan) = *last_scan {
+            if last_scan.elapsed().as_secs() < 60 {
+                // For the first day after installation, we only debounce to 2s:
+
+                let configure_time = context.get_config_i64(Config::ConfiguredTimestamp).await;
+                if time() - configure_time > 24 * 60 * 60 || last_scan.elapsed().as_secs() < 2 {
```

Comment:
What is the rationale behind this debouncing logic? Why the timer is different for the first day?

Reply:
Because maybe after the first day the spam filter is trained not to regard DC messages as spam. No idea how much sense this makes or whether servers can even be trained by moving over IMAP (and, if mvbox_move is on, not even to the inbox but to some random folder), but we might find out if a user complains that after a day messages arrive with a 60s delay :)

The other reason was that the next step is:
> Later on, after the above works, we can consider heuristics on which folders to perform IDLE on, and remove the Watch-folder options (inbox, deltachat, sent). We tried to find a safe scheme for already doing it but failed to fine one, too many unknowns, also some questions regarding multi-device (you might have different settings with each of it, one moves, the other doesn't etc.) so we postponed this in favor of the above incremental suggestion

and after a day we might know in which folders messages arrive.

Reply:
I would just debounce it to 1 day from the start then, but this will break the tests. Let's say this fast rescan for the first day is to simplify testing.